freetextresponse/public/view.js

```diff
@@ -139,4 +144,24 @@ function FreeTextResponseView(runtime, element) {
         userAlertMessage.text('');
         setClassForTextAreaParent('unanswered');
     });
+
+    $xblocksContainer.ready(function() {
+        $.ajax(url, {
+            type: 'POST',
+            data: JSON.stringify({
+                'student_answer': $element.find('.student_answer').val()
+            }),
+            success: function xblockLoadOnSuccess(response) {
+                display_responses_if_answered(response);
```

Comment:
I'm not sure if refactoring those two would net us any benefit right now. There are a bunch of differences both in data being sent, as well as in success reactions. (I already factored out the overlapping stuff into a helper)

Unfortunately we CAN'T do the AJAX request any earlier, XBlocks get loaded AFTER page load. Since I need to query the XBlock to get its answer, as well as peer responses, I HAVE to wait for the XBlock to be loaded like I do...

Reply:
> Unfortunately we CAN'T do the AJAX request any earlier

I think you may have missed my point:

> we should just include them directly in the original page load/response.

I'm suggesting that you _not_ use AJAX _at_ _all_ for initializing the XBlock.

Reply:
The problem is that `response.other_responses` and the `response.display_other_responses` setting value are only available in an AJAX response... so we CAN'T know if/what we should be displaying without an AJAX rrequest.

Reply:
ok, I guess we could, preload the HTML with answers, but I'm actually not sure if we should?